matrix-sdk/src/main/java/org/matrix/androidsdk/rest/callback/RestAdapterCallback.java

```diff
@@ -130,33 +145,36 @@ public void success(T t, Response response) {
         }
     }
 
+
     /**
      * Default failure implementation that calls the right error handler
-     * @param error the retrofit error
      */
-    @Override
-    public void failure(RetrofitError error) {
+    public void failure(Response<T> response, Exception exception) {
         if (null != mEventDescription) {
-            Log.d(LOG_TAG, "## failure(): [" + mEventDescription + "]" + " with error " + error.getMessage());
+            String error = exception != null
+                ? exception.getMessage()
+                : (response != null ? response.message() : "unknown");
+
+            Log.d(LOG_TAG, "## failure(): [" + mEventDescription + "]" + " with error " + error);
         }
 
         boolean retry = true;
 
-        if (null != error.getResponse()) {
-            retry = (error.getResponse().getStatus() < 400) || (error.getResponse().getStatus() > 500);
+        if (null != response) {
+            retry = (response.code() < 400) || (response.code() > 500);
         }
 
         if (retry && (null != mUnsentEventsManager)) {
             Log.d(LOG_TAG, "Add it to the UnsentEventsManager");
-            mUnsentEventsManager.onEventSendingFailed(mEventDescription, mIgnoreEventTimeLifeInOffline, error, mApiCallback, mRequestRetryCallBack);
+            mUnsentEventsManager.onEventSendingFailed(mEventDescription, mIgnoreEventTimeLifeInOffline, response, exception, mApiCallback, mRequestRetryCallBack);
         } else {
-            if (error.isNetworkError()) {
+            if (exception != null && exception instanceof IOException) {
                 try {
                     if (null != mApiCallback) {
                         try {
-                            mApiCallback.onNetworkError(error);
+                            mApiCallback.onNetworkError((Exception) exception);
```

Comment:
`exception` is already an instance of `Exception` class, why do you try to cast it?